####异常是一种exception类型的对象
1.有风险\会抛出异常的程序代码
 ```java
public void takeRisk() throws BadException {
    if(abandonAllHope) {
        throw new BadException();
    }
}
```
2.调用该方法的程序代码
```java
public void crossFingers() {
    try{
        anObject.takeRisk();
    } catch (BadException ex) {
        ex.printStackTrace();
    }
}
```

####异常处理规则
* catch与finally不能没有try
```java
void go() {
    Foo f = new Foo();
    f.foof();
    catch(FooException ex) { }
}
```  
* try一定要有catch或finally
```java
try {
    x.doStuff();
} finally {
    //清理
}
```
* try与catch之间不能有程序
```java
try {
    x.doStuff();
}
//下面这行不应存在
int y = 43;
catch(Exception ex) {
    
}
```
* 只带有finally的try 必须要声明异常
```java
void go() throws FooException {
    try{
        x.doStuff();
    } finally {
        
    }
}
```




